tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Wcf/TaskMethodInvokerIntegration.cs

```diff
@@ -0,0 +1,90 @@
+// <copyright file="TaskMethodInvokerIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#if NETFRAMEWORK
+using System;
+using System.ComponentModel;
+using System.Reflection;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.ClrProfiler.Emit;
+using Datadog.Trace.ClrProfiler.Integrations;
+using Datadog.Trace.Configuration;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Wcf
+{
+    /// <summary>
+    /// System.ServiceModel.Dispatcher.TaskMethodInvoker calltarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "System.ServiceModel",
+        TypeName = "System.ServiceModel.Dispatcher.TaskMethodInvoker",
+        MethodName = "InvokeAsync",
+        ReturnTypeName = "System.Threading.Tasks.Task`1<System.Tuple`2<System.Object, System.Object[]>>",
+        ParameterTypeNames = new[] { ClrNames.Object, "System.Object[]" },
+        MinimumVersion = "4.0.0",
+        MaximumVersion = "4.*.*",
+        IntegrationName = IntegrationName)]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class TaskMethodInvokerIntegration
+    {
+        private const string IntegrationName = nameof(IntegrationIds.Wcf);
+        private static readonly Func<object> _getCurrentOperationContext;
+
+        static TaskMethodInvokerIntegration()
+        {
+            var operationContextType = Type.GetType("System.ServiceModel.OperationContext, System.ServiceModel, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089", throwOnError: false);
+            if (operationContextType is not null)
+            {
+                var property = operationContextType.GetProperty("Current", BindingFlags.Public | BindingFlags.Static);
+                var method = property.GetGetMethod();
+                _getCurrentOperationContext = (Func<object>)method.CreateDelegate(typeof(Func<object>));
+            }
```

Comment:
Given this is the same (I think) in all the new integrations, is it worth moving to a helper, and using a generic helper with `<TTarget>` to cache it between all the instances)? We could also avoid the reflection entirely if the integration or the feature flag is disabled?

Reply:
haha @andrewlock also notice it... I agree.

Reply:
Yeah I got too excited about posting the PR that I forgot to consolidate the methods. Done in c5987de0a5f7c8a8af712feb71a87c9f9347303e